src/ast/rewriter/bool_rewriter.cpp

```diff
@@ -293,7 +294,8 @@ br_status bool_rewriter::mk_nflat_or_core(unsigned num_args, expr * const * args
 #endif
         if (s) {
```

Comment:
I think the code should be

```
        if (m_sort_disjunctions && s) {
            ast_lt lt;
            std::sort(buffer.begin(), buffer.end(), lt);       
            result = m().mk_or(sz, buffer.data());
            return BR_DONE;
        }
```

If nothing is sorted, then the rewrite has no effect. The rewrite engine is saved from rebuilding expressions.